rotkehlchen/chain/evm/decoding/merkl/utils.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+from rotkehlchen.errors.misc import RemoteError, UnableToDecryptRemoteData
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.cache import globaldb_set_general_cache_values
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import CacheType, ChainID
+from rotkehlchen.utils.network import request_get
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _paginated_request(url_template: str, items_per_page: int, context: str, **url_params: Any) -> list | None:  # noqa: E501
+    """Make paginated requests to Merkl API.
+
+    `url_template` is the URL template with {page} and {items} placeholders
+    `items_per_page` is the number of items per page
+    `context` is the context string for error logging
+    `**url_params` are the additional parameters for URL formatting
+
+    Returns a list of all items from all pages, or None if there was an error.
+    """
+    all_items: list[Any] = []
+    page = 0
+    while True:
+        try:
+            url = url_template.format(page=page, items=items_per_page, **url_params)
+            response = request_get(url)
+
+            if len(response) == 0:
+                break  # No more items to fetch
+
+            all_items.extend(response)
+
+            # If we got less than the full page size, we've reached the end
+            if len(response) < items_per_page:
+                break
```

Comment:
actually

```suggestion
            all_items.extend(response := request_get(url))
            # If we got less than the full page size, we've reached the end
            if len(response) == 0 or len(response) < items_per_page:
                break
```

Reply:
and actually the 0 case is inside `len(response) < items_per_page:` so you don't need it